Added txtConflict logic (#24)
Domain-Connect/spec: Domain Connect Spec Draft.adoc

|String
|groupId
|(optional)
This parameter identifies the group the record belongs to when applying changes. This cannot contain
variables.

|[[essential-record]]*Essential*
|enum
|essential
|(optional)
This parameter indicates how the record should be treated during conflict detection (if the DNS Provider
is not implementing Conflict Detection it is ignored).

Always (default) - record MUST be applied and kept with the template

OnApply - record MUST be applied but can be later removed without dropping the whole
template

|*Host*
|String|host a|
The host for A, AAAA, CNAME, TXT, and MX values.

This is the hostname in DNS. It is heavily recommended that this not contain variables.
However, this may be necessary and is discussed below.

|[[pointsto-record]]*Points To*
|String
|pointsTo
|The pointsTo location for A, AAAA, CNAME, and MX records.

|*TTL*
|Int
|ttl
|The time-to-live for the record in DNS. Valid
for A, AAAA, CNAME, TXT, MX, and SRV records. As a matter of practice this does not contain
variables.

|*Data*
|String
|data
|The data for a TXT record in DNS

|*TXT Conflict Matching Mode*
|String
|txtConflictMatchingMode

> |Describes how conflicts on the TXT record should be handled. Possible values are 

Author: Done

> |Describes how conflicts on the TXT record should be handled. Possible values are 

Reviewer: Done

None, All, or Prefix. The default value is None. See below.

|*TXT Conflict Matching Prefix*
|String
|txtConflictMatchingPrefix

> |The prefix for handling deletes when txtConflictMatchingMode is "Prefix". This 

Author: Done

> |The prefix for handling deletes when txtConflictMatchingMode is "Prefix". This 

Reviewer: Done

this would be services that would be disabled. For records, this would be
records that would be overwritten. This could be progressively disclosed.

Note: When re-applying the same template, ideally DNS Providers would not detect
the conflict. Instead, the first template would be removed and the new
instance applied. For most templates this is a benign operation, unless
the template contains variables in host names. For consideration
of this, see the section below.

> === Record Types and Conflicts

Reviewer: Reading through the changes I now notices, that the whole section is talking about deleting or adding records, while it should be rather referring to detection of situations with or without conflict.

That the conflict resolution leads to deletion of the conflicting records upon new template application is something we describe in the previous chapter, also in a bit vague way. I'll issue a proposal to rephrase the whole thing...

Reviewer: Not actual anymore

Author: I think you re-wrote this.

The proposed handling of records and conflicts is as follows (if not
otherwise specified, conflicts occur if the records have the same name):

* Replace records of the same type for A, AAAA, MX, CNAME, APEXCNAME,
SRV. If the template specifies an A or AAAA, the respective A or AAAA
record should be removed to avoid IPv4 and IPv6 pointing to different
services.
* For TXT records deletion is handled looking at txtConflictMatchingMode

> ** None: This indicates that the TXT records should be appended to the zone without removing existing values of the same host.

Reviewer: ```suggestion
** None: This indicates that the TXT records should be appended to the zone without removing existing values. This is the default behavior.```

Reviewer: Not actual anymore

> ** All: This indicates that the TXT records with the same host name are deleted prior to applying.

Reviewer: ```suggestion
** All: This indicates that all the TXT records are deleted prior to applying.
```

Reviewer: Not actual anymore
